chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/chartdata/ParticipantAgeStatisticsCalculator.java

```diff
@@ -99,22 +94,16 @@ private void initialize() {
     }
 
     private void save(final List<ParticipantAgeStatisticsEntity> entities) {
```

Comment:
Shouldn't this method be @Transactional too? This is the method that does the delete & save calls, right? So this should be done as a single transaction

Reply:
The job's execute method is annotated as transactional and I think Spring's default behavior is to join an existing transaction if there is one otherwise start a new one... so all these transactional methods (I think) will be part of the top-level transaction. If that's not desired then some re-organization of the annotations should be done. Generally we try not to make DAO methods transactional but leave it up to the user of the DAO (a manager or calculator or something) to determine what should happen atomically.

Reply:
The reason I added those Transactional tags to the DAO methods is becuase I was getting errors like the following "javax.persistence.TransactionRequiredException: no transaction is in progress" when the entityManager.flush(); methods were getting called within those methods, in the ListingCountStatisticsDAOImpl for example. Now, Im not entirely sure why those transactional annotations worked but I just tried to take those annotations away from the DAO methods and added it to the top level save method inside ListingCountsStatisticsCalculator and got the same error as above.

Reply:
Ok, I don't think I understand how all of the Spring stuff works in the jobs vs how it works in our webapp classes because I thought none of that autowired stuff should be working either when classes are created with new Something() but it works. Maybe @tmy1313  has some insight? Or just leave the annotations and forget about it since it's working?

Reply:
I'm fine with moving forward with this as well!